### PR TITLE
modify LabelsEqual for unused_deps and some tests

### PR DESCRIPTION
**Pull request by shvar**
@pmbethe09, I modified a bit LabelsEqual in order to deal with `.jar-2.params` files notation.

Assuming, that I have a `maven_jar` defined in a WORKSPACE like:
```
maven_jar(
    name = "org_apache_commons_commons_lang3",
    artifact = "org.apache.commons:commons-lang3:3.1",
    sha1 = "905075e6c80f206bbe6cf1e809d2caa69f420c76",
)
```
for every direct dep on it in BUILD file I got in params file following string: `@@org_apache_commons_commons_lang3//jar:jar`

I have two problems with this
1) leading `@@`, that I think should be trimmed (and that's the reason of modifying LabelEquals)
2) `:jar` suffix, that is usually omit in BUILD files. Here, I am not sure about what to do and lean to force administratively demand writing full label name with :jar extension.

WDYT?

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by shvar** [comment]
I signed it!

**Comment by pmbethe09** [comment]
Jenkins, test this please.

**Comment by pmbethe09** [comment]
Is @@ even legal in bazel?  Where does that come from?

**Comment by shvar** [comment]
I don't think, it's a legal name for bazel label, but I see them in bazel-bin/libruleName.jar-2.params. Do you think it is actually a bug somewhere in the code, that generates that file?

**Comment by pmbethe09** [comment]
That does sound like a bug -- @damienmg any ideas why the .jar-2.params would have a double @ in it?

**Comment by damienmg** [comment]
Yes because @XXX is the syntax for params file for javabuilder. @@ is the way to escape that, we should de-dup the @@ when reporting in jdeps. /cc @cushon

**Comment by cushon** [comment]
@damienmg I sent you a fix.

**Comment by damienmg** [comment]
Wow thanks for the fast turn around!

**Comment by shvar** [comment]
Wow, that was fast, thank you all!

So, @pmbethe09, I reverted logic of trimming @@ in favor of @cushon 's fix. Now it makes even more sense to use LabelsEqual in unused_deps.

File: edit/edit_test.go [hunk 1]
```diff
@@ -67,6 +67,26 @@ func TestShortenLabel(t *testing.T) {
 	}
 }
 
+var labelsEqualTests = []struct {
+	label1 string
+	label2 string
+	pkg    string
+	result bool
```
Comment (pmbethe09) [inline edit/edit_test.go]: in our Go-style this is often called 'want' or 'expected

File: edit/edit_test.go [hunk 2]
```diff
@@ -67,6 +67,26 @@ func TestShortenLabel(t *testing.T) {
...
+}{
+	{"//devtools/buildozer:rule", "rule", "devtools/buildozer", true},
+	{"//devtools/buildozer:rule", "rule:jar","devtools", false},
+}
+
+func TestLabelsEqual(t *testing.T) {
+	for i, tt := range labelsEqualTests {
+		result := LabelsEqual(tt.label1, tt.label2, tt.pkg)
```
Comment (pmbethe09) [inline edit/edit_test.go]: s/result/got/  (our Go-style for tests)

**Comment by pmbethe09** [comment]
Jenkins, test this please.

File: edit/edit_test.go (hunk 2 above)
Comment (shvar) [inline edit/edit_test.go]: sure, fixed both

File: unused_deps/unused_deps.go [hunk 3]
```diff
@@ -221,11 +221,11 @@ func printCommands(label string, deps map[string]bool) (anyCommandPrinted bool)
 		for _, elem := range li.List {
 			for dep := range deps {
 				str, ok := elem.(*build.StringExpr)
-				if ok && edit.ShortenLabel(str.Value, pkg) == edit.ShortenLabel(dep, pkg) {
+				if ok && edit.LabelsEqual(str.Value, dep, pkg) {
```
Comment (pmbethe09) [inline unused_deps/unused_deps.go]: now that bazel is fixing the '@@' escaping I have forgotten why this change is necessary anymore?

Comment (shvar) [inline unused_deps/unused_deps.go]: Now it's more about cosmetics and using in buildozer exact label, as specified in BUILD file.

**Comment by pmbethe09** [comment]
Jenkins, test this please.
